File: storage.js
const fs = require('fs');

function saveData(data, path='data.json') {
    fs.writeFileSync(path, JSON.stringify(data, null, 2));
}

function loadData(path='data.json') {
    try {
        return JSON.parse(fs.readFileSync(path, 'utf8'));
    } catch (error) {
        if (error.code === 'ENOENT') {
            return [];
        } else {
            throw error;
        }
    }
}

module.exports = { saveData, loadData };

File: storage.py
import json
import os

def save_data(data, path='data.json'):
    """Save data to a JSON file."""
    with open(path, 'w') as file:
        json.dump(data, file, indent=2)

def load_data(path='data.json'):
    """Load data from a JSON file."""
    if os.path.exists(path):
        with open(path, 'r') as file:
            return json.load(file)
    else:
        return []
